src/semidiscretization/semidiscretization_coupled.jl

```diff
@@ -435,10 +435,25 @@ function (boundary_condition::BoundaryConditionCoupled)(u_inner, orientation, di
                                 Val(nvariables(equations))))
 
     # Calculate boundary flux
-    if iseven(direction) # u_inner is "left" of boundary, u_boundary is "right" of boundary
-        flux = surface_flux_function(u_inner, u_boundary, orientation, equations)
-    else # u_boundary is "left" of boundary, u_inner is "right" of boundary
-        flux = surface_flux_function(u_boundary, u_inner, orientation, equations)
+    if surface_flux_function isa Tuple
+        # In case of conservative and non-conservative fluxes add the non-conservative one with a factor of 1/2.
+        if iseven(direction) # u_inner is "left" of boundary, u_boundary is "right" of boundary
+            flux = surface_flux_function[1](u_inner, u_boundary, orientation,
+                                            equations) +
+                   0.5 *
+                   surface_flux_function[2](u_inner, u_boundary, orientation, equations)
+        else # u_boundary is "left" of boundary, u_inner is "right" of boundary
+            flux = surface_flux_function[1](u_boundary, u_inner, orientation,
+                                            equations) +
+                   0.5 *
+                   surface_flux_function[2](u_boundary, u_inner, orientation, equations)
+        end
```

Comment:
```suggestion
        if iseven(direction) # u_inner is "left" of boundary, u_boundary is "right" of boundary
            flux = (surface_flux_function[1](u_inner, u_boundary, orientation,
                                            equations) +
                   0.5 *
                   surface_flux_function[2](u_inner, u_boundary, orientation, equations))
        else # u_boundary is "left" of boundary, u_inner is "right" of boundary
            flux = (surface_flux_function[1](u_boundary, u_inner, orientation,
                                            equations) +
                   0.5 *
                   surface_flux_function[2](u_boundary, u_inner, orientation, equations))
        end
```
Add parentheses for safety. Also the auto-formatting here is a bit ugly (just a comment really, not anything you need to fix).